### DISCUSSION TITLE
count of data from relation

### DISCUSSION THREAD

**Discussion by NargizMe** [supabase/supabase#10530] [Questions]
Hi , i could't find info in web so decided to ask here. I have a query and i want also get all comments length )

```
let { data, error } = await supabase
      .from("posts")
      .select(`*, users(name, image), comments(*, users(name, image)), liked(user_id)`)
      .eq('liked.user_id', userId)
      .order('created_at', {
        foreignTable: 'comments',
        ascending: false
      })
      .order('created_at', {
        ascending: false
      })
      .limit(3, {
        foreignTable: 'comments'
      })
      .range(from, to);
```

After this query i get 3 comments which is what i want , but i cant figure out how to also get all comments count from table.

**Comment by GaryAustin1** [comment]
You might be able to just add count in the comments part of the select.

```
.select(`*, users(name, image), comments(*, users(name, image), count), liked(user_id)`)
```

**Reply by NargizMe** [reply]
i am getting this kind of error

```
{
  "code":"42803",
  "details":null,
  "hint":null,
  "message":"column \"comments.id\" must appear in the GROUP BY clause or be used in an aggregate function"
}
```

I have searched about grouping and i found that i have to create table with custom postgress function or something like that, is it true ?

**Reply by GaryAustin1** [reply]
Not sure.  
This is an example from another discussion here. Just search on count. , there are many.  

https://github.com/supabase/supabase/discussions/1113#discussioncomment-594673

number_of_items:items(count)  

Maybe you have to just do the count separately but I don’t think so.

**Reply by steve-chavez** [reply]
Ah, yes, this is a limitation for now. You can only use count as a single column on the related table.

```js
.select(`*, users(name, image), comments(count), liked(user_id)`)
```

Otherwise it will give the above [error](https://github.com/supabase/supabase/discussions/10530#discussioncomment-4242008).

> I have searched about grouping and i found that i have to create table with custom postgress function or something like that, is it true ?

If you need more complex queries with explicit GROUP BY then yes, you'd have to use [rpc](https://supabase.com/docs/reference/javascript/rpc).

**Reply by NargizMe** [reply]
> Not sure. This is an example from another discussion here. Just search on count. , there are many.
> 
> [#1113 (comment)](https://github.com/supabase/supabase/discussions/1113#discussioncomment-594673)
> 
> number_of_items:items(count)
> 
> Maybe you have to just do the count separately but I don’t think so.

Thank you very much. Actually it worked.
This is the working version of select.

`      .select('*, users(name, image), comments(*, users(name, image)), number_of_comments:comments(count), liked(user_id)')
`
